pkg/clusteragent/autoscaling/cluster/model/node_pool_internal.go

```diff
@@ -214,17 +210,55 @@ func buildNodePoolSpec(n NodePoolInternal, nodeClassName string) karpenterv1.Nod
 		},
 	}
 
-	if n.TargetWeight() != nil {
-		targetWeight := *n.TargetWeight()
+	return npSpec
+}
+
+// BuildReplicaNodePool updates the target NodePool spec to create a replica NodePool
+// TODO: Add logic for any existing requirements that could be incompatible with recommendations
+func BuildReplicaNodePool(knp *karpenterv1.NodePool, npi NodePoolInternal) {
+
+	// Update NodePool with recommendation
+	instanceTypeLabelFound := false
+	for _, r := range knp.Spec.Template.Spec.Requirements {
+		if r.NodeSelectorRequirement.Key == corev1.LabelInstanceTypeStable {
+			r.Operator = corev1.NodeSelectorOpIn
+			r.Values = npi.RecommendedInstanceTypes()
+
+			instanceTypeLabelFound = true
+		}
+	}
+	if !instanceTypeLabelFound {
+		knp.Spec.Template.Spec.Requirements = append(knp.Spec.Template.Spec.Requirements,
+			karpenterv1.NodeSelectorRequirementWithMinValues{
+				NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+					Key:      corev1.LabelInstanceTypeStable,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   npi.RecommendedInstanceTypes(),
+				},
+			},
+		)
+	}
+
+	// Update NodePool with weight
+	if npi.TargetWeight() != nil {
```

Comment:
if the `targetWeight` is `nil`, do we set a default weight for nodepools we create? or would the replica also have a weight of nil (and in that case how do we guarantee that ours is taken over the existing nodepool?)